# Superstore Sales Analysis

**Author:** Eman Mahmoud
**Date:** 2026-09-16
**Dataset:** Superstore Sales (Kaggle)

## Project Objective
This project analyzes Superstore sales to identify which products, regions,
and customer segments drive the most revenue and profit, and to flag areas
where sales and profit diverge — specifically, where high sales come with
low or negative profit.

## Analysis Questions
1. Which product categories and regions generate the most revenue and profit?
2. How have sales and profit changed over time (monthly/yearly)?
3. How does discount level relate to profit?

## Tools
Python, pandas, matplotlib, seaborn, sqlite3

Importing required libraries.

In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import sqlite3

sns.set_style("whitegrid")
%matplotlib inline

Loading the raw dataset for initial inspection (no cleaning yet).

In [5]:
df = pd.read_csv("../data/superstore.csv", encoding="latin-1")
print(df.shape)
print(df.dtypes)
print(df.head())
print(df.isnull().sum())
print(df.describe(include="all"))

(9994, 21)
Row ID             int64
Order ID             str
Order Date           str
Ship Date            str
Ship Mode            str
Customer ID          str
Customer Name        str
Segment              str
Country              str
City                 str
State                str
Postal Code        int64
Region               str
Product ID           str
Category             str
Sub-Category         str
Product Name         str
Sales            float64
Quantity           int64
Discount         float64
Profit           float64
dtype: object
   Row ID        Order ID  Order Date   Ship Date       Ship Mode Customer ID  \
0       1  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
1       2  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
2       3  CA-2016-138688   6/12/2016   6/16/2016    Second Class    DV-13045   
3       4  US-2015-108966  10/11/2015  10/18/2015  Standard Class    SO-20335   
4       5  US-2015-108966  10/11/2015  10/18/

In [6]:
print(pd.to_datetime(df["Order Date"]).min())
print(pd.to_datetime(df["Order Date"]).max())

2014-01-03 00:00:00
2017-12-30 00:00:00


## Initial Data Inspection Notes

- **Rows and columns:** The dataset has 9,994 rows and 21 columns.
- **Column types:** 
  - Numeric: `Row ID`, `Postal Code`, `Sales`, `Quantity`, `Discount`, `Profit`
  - Categorical (text): `Order ID`, `Ship Mode`, `Customer ID`, `Customer Name`,
    `Segment`, `Country`, `City`, `State`, `Region`, `Product ID`, `Category`,
    `Sub-Category`, `Product Name`
  - Date (currently stored as text, will need conversion): `Order Date`, `Ship Date`
- **Missing values:** None. All 21 columns show 0 missing values.
- **Date range:** Orders span from 2014-01-03 to 2017-12-30 (4 years).

# Analysis Questions

## Question 1
**Question:** Which product categories and regions generate the most revenue and profit?
**Why it matters:** Identifying top-performing categories and regions helps
the business decide where to focus inventory, marketing, and sales effort.
It also reveals whether any category or region drives high sales but low profit.
**Chart I expect to use:** Bar chart (sales/profit by category and by region).

## Question 2
**Question:** How have sales and profit changed over time (monthly/yearly)?
**Why it matters:** Understanding the trend over the 4-year period shows
whether the business is growing, shrinking, or seasonal, which affects
forecasting and planning decisions.
**Chart I expect to use:** Line chart (sales/profit over time).

## Question 3
**Question:** How does discount level relate to profit?
**Why it matters:** Discounts drive sales volume but can erode margins.
Understanding this relationship helps the business set discount policy
without unintentionally destroying profit.
**Chart I expect to use:** Scatter plot (discount % vs. profit), possibly
supported by a correlation coefficient.